uqcsbot/utils/uq_course_utils.py

```diff
@@ -52,12 +54,27 @@ def __init__(self, url, status_code):
         self.status_code = status_code
         super().__init__(self.message, self.url, self.status_code)
 
+def get_offering_code(semester=None, campus='STLUC', is_internal=True):
+    """
+    Returns the hex encoded offering string for the given semester and campus.
+
+    Keyword Arguments:
+        semester {str} -- Semester code (3 is summer) (default: current semester)
+        campus {str} -- Campus code string (one of STLUC, etc.)
+        is_internal {bool} -- True for internal, false for external.
+    """
+    # TODO: Codes for other campuses.
+    if semester is None:
+        semester = 1 if datetime.today().month <= 6 else 2
+    location = 'IN' if is_internal else 'EX'
+    return hexlify(f'{campus}{semester}{location}'.encode('utf-8')).decode('utf-8')
+
 
 def get_course_profile_url(course_name):
     '''
     Returns the URL to the latest course profile for the given course.
     '''
-    course_url = BASE_COURSE_URL + course_name
+    course_url = BASE_COURSE_URL + course_name + OFFERING_QUERY + get_offering_code()
```

Comment:
Since we now have more than one param, I would suggest using requests `params` as seen here http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls :^)